dissect/target/filesystems/xfs.py

```diff
@@ -130,8 +130,14 @@ def lstat(self) -> fsutil.stat_result:
         st_info.st_mtime_ns = self.entry.mtime_ns
         st_info.st_ctime_ns = self.entry.ctime_ns
 
-        # XFS has a birth time, called crtime
+        st_info.st_blksize = self.fs.xfs.block_size
+        # Convert number of filesystem blocks to basic blocks
+        # Reference: https://github.com/torvalds/linux/blob/e32cde8d2bd7d251a8f9b434143977ddf13dcec6/fs/xfs/xfs_iops.c#L602 # noqa: E501
+        st_info.st_blocks = self.entry.nblocks * (self.fs.xfs.block_size // 512)
```

Comment:
```suggestion
        st_info.st_blocks = self.entry.nblocks * (self.fs.xfs.block_size // 512)
```

This does premature rounding and also rounds down. The POSIX/opengroup documentation of `st_blocks` has the comment: "Number of blocks allocated for this object.". The block size in our implementation is 512 bytes.

A correct implementation would be:

`math.ceil(self.entry.nblocks * self.fs.xfs.block_size / 512)`

I'm not sure if `nblocks` is 0 for directories, otherwise the st_blocks for dirs should be special cased to 0. The same holds for symlinks as long as their data is just contained inside the inode/filsystem structure.

Reply:
@Schamper 

I did the following experiment

1. I created a path of size 4096 to file A. This forces a non-residential symlink.
2. I created a symlink to file A.
3. I verified that the symlink is non-resident by using `xfs_db` and checking `core.format`==2. 
4. I checked `core.nblocks`. It is zero. Stat on linux also returns zero. Stat on target-shell returns zero.

Reply:
I found an issue with a new symbolic link... investigating